topics/statistics/tutorials/classification_machinelearning/tutorial.md

```diff
@@ -0,0 +1,530 @@
+
+
+# Random Forest
+
+[Random forest](https://en.wikipedia.org/wiki/Random_forest) is an ensemble of decision trees, and usually trained with the “bagging” method. [Ensemble](https://scikit-learn.org/stable/modules/ensemble.html#ensemble) method uses multiple learning models internally for better predictions and the general idea of the bagging method is that a combination of learning models increases the overall result. It uses multiple decision tree regressors internally and predicts by taking the collective performances of the predictions by multiple decision trees. It has a good predictive power and is robust to the outliers. It creates an ensemble of weak learners (decision trees) and iteratively minimizes error. One big advantage of random forest is that it can be used for both classification and regression problems. The main idea behind the random forest is adding additional randomness to the model, while growing the trees and instead of searching for the most important feature while splitting a node, it searches for the best feature among a random subset of features. This results better model because of wide diversity. Generally, the more trees in the forest the more robust the forest looks like. In the same way in the random forest classifier, the higher the number of trees in the forest gives the high accuracy results. Simlarly there are two stages in random forest algorithm, one is random forest creation, the other is to make a prediction from the random forest classifier created in the first stage.
+
+> ### {% icon hands_on %} Hands-on: Train random forest
+>
+> 1. **Ensemble methods** {% icon tool %} with the following parameters to train the regressor:
+>    - *"Select a Classification Task"*: `Train a model`
+>       - *"Select an ensemble method"*: `Random forest classifier`
+>          - *"Select input type"*: `tabular data`
+>             - {% icon param-file %} *"Training samples dataset"*: `train_rows.csv`
+>             - {% icon param-check %} *"Does the dataset contain header"*: `Yes`
+>             - {% icon param-select %} *"Choose how to select data by column"*: `All columns EXCLUDING some by column header name(s)`
+>                - {% icon param-text %} *"Type header name(s)"*: `Class`
+>             - {% icon param-file %} *"Dataset containing class labels"*: `train_rows.csv`
+>             - {% icon param-check %} *"Does the dataset contain header"*: `Yes`
+>             - {% icon param-select %} *"Choose how to select data by column"*: `Select columns by column header name(s)`
+>                - {% icon param-text %} *"Select target column(s)"*: `Class`
+> 2. Rename the generated file to `RandomForest_model`
+{: .hands_on}
+
+> ### {% icon question %} Question
+>
+> What are the advantages of random forest classifier compare with classifiers?
+>
+> > ### {% icon solution %} Solution
+> > 1. The overfitting problem will never come when we use the random forest algorithm in any classification problem.
```

Comment:
How are you sure about it? Would you like to give a reason?

Reply:
In section 2 of Breiman's paper (https://link.springer.com/article/10.1023/A:1010933404324) which proposes Random Forest, there is proof. "Use of the Strong Law
of Large Numbers shows that they always converge so that overfitting is not a problem."

I have changed the sentence a bit.